includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -702,10 +702,14 @@ public function prepare_order_source( $order = null ) {
 				$source_object = WC_Stripe_API::retrieve( 'sources/' . $source_id );
 			} elseif ( apply_filters( 'wc_stripe_use_default_customer_source', true ) ) {
 				/*
-				 * We can attempt to charge the customer's default source
-				 * by sending empty source id.
+				 * ~We can attempt to charge the customer's default source~
+				 * ~by sending empty source id.~
+				 * We retrieve the default customer source manually.
 				 */
-				$stripe_source = '';
+				$default_source = $this->get_default_customer_source_for_order( $order );
```

Comment:
Do we know that this change works for all cases where `prepare_order_source` is used? It's two more: pre-orders and SEPA subs.

Reply:
> Do we know that this change works for all cases where prepare_order_source is used? It's two more: pre-orders and SEPA subs.

I did a check with Pre-Orders and it worked, however SEPA subs are a mixed bag. If there is no source entered, the code will fall back to the default credit card without using PaymentIntents. I'm ready to open a separate issue about this, but I'm not sure how to approach it. Should we simply select a source from the particular/needed type?